extensions/wikia/Forum/Forum.class.php

```diff
@@ -334,29 +332,41 @@ public function deleteBoard( $board ) {
 		wfProfileOut( __METHOD__ );
 	}
 
-	public function createDefaultBoard() {
-		wfProfileIn( __METHOD__ );
-		$app = F::App();
-		if ( !$this->hasAtLeast( NS_WIKIA_FORUM_BOARD, 0 ) ) {
-			WikiaDataAccess::cachePurge( wfMemcKey( 'Forum_hasAtLeast', NS_WIKIA_FORUM_BOARD, 0 ) );
-			/* the wgUser swap is the only way to create page as other user then current */
-			$tmpUser = $app->wg->User;
-			$app->wg->User = User::newFromName( Forum::AUTOCREATE_USER );
+	/**
+	 * Check if there are any forum boards. If not: create the 5 default one using user from Forum::AUTOCREATE_USER
+	 *
+	 * Note this function returns true even if the boards were not successfully created.
+	 *
+	 * @return bool true if tried creating the boards, false if there were some boards already
+	 */
+	public function createDefaultBoards() {
+		if ( $this->hasMoreThan( NS_WIKIA_FORUM_BOARD, 0 ) ) {
+			return false;
+		}
+
+		WikiaDataAccess::cachePurge( wfMemcKey( 'Forum_hasMoreThan', NS_WIKIA_FORUM_BOARD, 0 ) );
```

Comment:
Shouldn't we purge the cache **after** the first board is created? This will prevent unlikely (but still possible) race conditions.